File: ReadyChef.Core/Models/Recipe.cs
using System;
using System.Collections.Generic;

namespace ReadyChef.Core.Models
{
    public class Recipe
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int ReadyTime { get; set; }

        public List<RecipeIngredient> RecipeIngredients { get; set; }
    }
}

File: ReadyChef.Api/Scripts/services/recipe.js
angular.module('readyChefApp')
    .service('recipe',recipe);
recipe.$inject=['recipeIngredientFactory'];
function recipe(recipeIngredientFactory){

    function makeNewRecipe(){
        var vm = this;
        vm.name = "";
        vm.recipeIngredients = [];
        vm.readyTime = "";
        vm.addRecipeIngredient = addRecipeIngredient;
        vm.id = "";

        function addRecipeIngredient(ingredientName, amount, amountUnits) {
            recipeIngredientFactory.makeNewRecipeIngredient(ingredientName, amount, amountUnits)
                .then(function (recipeIngredient) {
                    vm.recipeIngredients.push(recipeIngredient);
                });
        };
    };

    return makeNewRecipe;
};